pkg/resourcetotelemetry/resource_to_telemetry.go

```diff
@@ -54,28 +64,59 @@ func WrapMetricsExporter(set Settings, exporter component.MetricsExporter) compo
 	if !set.Enabled {
 		return exporter
 	}
-	return &wrapperMetricsExporter{MetricsExporter: exporter}
+	return &wrapperMetricsExporter{MetricsExporter: exporter, MatchAttributes: set.Exclude}
 }
 
-func convertToMetricsAttributes(md pmetric.Metrics) pmetric.Metrics {
+func convertToMetricsAttributes(md pmetric.Metrics, filter MatchAttributes) pmetric.Metrics {
 	cloneMd := pmetric.NewMetrics()
 	md.CopyTo(cloneMd)
 	rms := cloneMd.ResourceMetrics()
 	for i := 0; i < rms.Len(); i++ {
 		resource := rms.At(i).Resource()
 
+		labelMap := extractLabelsFromResource(&resource, filter)
+
 		ilms := rms.At(i).ScopeMetrics()
 		for j := 0; j < ilms.Len(); j++ {
 			ilm := ilms.At(j)
 			metricSlice := ilm.Metrics()
 			for k := 0; k < metricSlice.Len(); k++ {
-				addAttributesToMetric(metricSlice.At(k), resource.Attributes())
+				addAttributesToMetric(metricSlice.At(k), labelMap)
 			}
 		}
 	}
 	return cloneMd
 }
 
+// extractAttributesFromResource extracts the attributes from a given resource and
+// returns them as a StringMap.
+func extractLabelsFromResource(resource *pcommon.Resource, filter MatchAttributes) pcommon.Map {
+	labelMap := pcommon.NewMap()
+	//var err error
+
+	if len(filter.Exclude) == 0 {
+		return resource.Attributes()
+	}
+
+	attrMatcher := filtermatcher.AttributesMatcher{}
+	attrMatcher, _ = filtermatcher.NewAttributesMatcher(filter.Filter, filter.Exclude)
+
+	resource.Attributes()
+
+	attributes := resource.Attributes()
+	attrMap := attributes.AsRaw()
+	for k := range attrMap {
+		v, _ := attributes.Get(k)
+		mapp := pcommon.NewMap()
+		mapp.PutStr(k, v.AsString())
+		if !attrMatcher.MatchAny(mapp) {
+			stringLabel, _ := attributes.Get(k)
+			labelMap.PutStr(k, stringLabel.AsString())
+		}
+	}
```

Comment:
I don't get what you are trying to do here. There are a few things that you can improve:

1. Use [Range](https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pcommon#Map.Range) for looping over a map
2. It seems like `stringLabel` and `v` would be the same
3. We should avoid creating an auxiliary `mapp`just to do the matching

Reply:
refactored as suggested